src/world/config/shops.ts

```diff
@@ -0,0 +1,80 @@
+import { logger } from '@runejs/logger/dist/logger';
+import { JSON_SCHEMA, safeLoad } from 'js-yaml';
+import { readFileSync } from 'fs';
+import { world } from '@server/game-server';
+import { Player } from '@server/world/mob/player/player';
+
+export enum ShopName {
+
+}
+export interface Shop {
+    identification: string;
+    name: string;
+    interfaceId: number;
+    items: ShopItems[];
+}
+
+interface ShopItems {
+    id: number;
+    name: string;
+    amountInStock: number;
+    price: number;
+}
+
+export function parseShops(): Shop[] {
+    try {
+        logger.info('Parsing shops...');
+
+        const shops = safeLoad(readFileSync('data/config/shops.yaml', 'utf8'), { schema: JSON_SCHEMA }) as Shop[];
+
+        if(!shops || shops.length === 0) {
+            throw 'Unable to read shops.';
+        }
+
+        logger.info(`${shops.length} shops found.`);
+
+        return shops;
+    } catch(error) {
+        logger.error('Error parsing shops: ' + error);
+        return null;
+    }
+}
+
+function findShop(identification: string): Shop {
+    for(let i = 0; i <= world.shops.length; i++) {
+        if(world.shops[i].identification === identification) return world.shops[i];
+    }
+    return undefined;
+}
+
+export function openShop(player: Player, identification: string, closeOnWalk: boolean = true): void {
```

Comment:
Prefer this to be it's own action as opposed to having it inside the shop config file